### PR TITLE
feat[added Library and custom playlist pages]

### PR DESCRIPTION
**Pull request by Radheshyam1234**

### CODE REVIEW

File: src/Components/ListingPages/Library/Library.css [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.library-page {
+  max-width: 75rem;
+  /* padding: 1rem 2rem; */
```
Comment (MarkVed17) [inline src/Components/ListingPages/Library/Library.css]: you may remove this.

**Review by MarkVed17 — APPROVED** [review]
Rest all looks good. 
You may proceed!

File: src/App.js [hunk 2]
```diff
@@ -81,6 +83,32 @@ export const App = () => {
             </PrivateRoute>
           }
         />
+        <Route
+          path="/library"
+          element={
+            <PrivateRoute>
```
Comment (t007rushi) [inline src/App.js:89]: rename it ```RequireAuth``` as per convention

File: src/Components/ListingPages/AllPlaylists.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  useEffect(() => {
+    let playlists = [state.likedVideos, state.watchHistory, state.watchLater];
+    setDefaultPlaylists(playlists);
+    setCustomPlaylists(state.playlists);
+  }, [state]);
+  return (
+    <div className="playlists-container display-flex flex-wrap justify-center">
+      {defaultPlaylists?.map((playlist) => {
+        return (
+          <Link to={`/${playlist.type}`}>
```
Comment (t007rushi) [inline src/Components/ListingPages/AllPlaylists.js:19]: ```key``` prop misssing

File: src/Components/ListingPages/AllPlaylists.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            <div className="playlist-box box-shadow ">
+              <span>{playlist.type}</span>
+              <div></div>
+            </div>
+          </Link>
+        );
+      })}
+
+      {customPlaylists?.map((playlist) => {
```
Comment (t007rushi) [inline src/Components/ListingPages/AllPlaylists.js:28]: suggested to destructure here ```{_id, name}```

File: src/Components/ListingPages/Library/Library.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    {
+      playlist: watchHistory,
+      path: "history",
+    },
+    {
+      playlist: watchLater,
+      path: "watchlater",
+    },
+  ];
```
Comment (t007rushi) [inline src/Components/ListingPages/Library/Library.js:25]: keep constant outside

File: src/Components/ListingPages/Library/Library.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+  return (
+    <div className="library-page">
+      {Playlists?.map((Playlist) => {
+        return (
+          <div key={Playlist.playlist._id}>
+            <p className="text-semibold">
+              {Playlist?.playlist?.type?.toUpperCase()} Videos (
+              {Playlist?.playlist?.videos?.length})
```
Comment (t007rushi) [inline src/Components/ListingPages/Library/Library.js:34]: same as above destructure for clean interpretation and clean codebase

File: src/Components/ListingPages/PlaylistVideos/PlaylistVideos.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+                <p>{playlist?.videos.length} videos </p>
+              </div>
+
+              <div className="display-flex">
+                <button
+                  className="btn secondary-btn"
+                  title=" Edit Title"
+                  onClick={() => {
+                    if (!token) navigate("/login");
```
Comment (t007rushi) [inline src/Components/ListingPages/PlaylistVideos/PlaylistVideos.js:87]: ?: ternary works here

**Review by t007rushi — APPROVED** [review]
Look for suggestions 
Naming is quite good 
Else after some minor changes can merge

**Pull request merged**